experimental/golden_response_checker.go

```diff
@@ -209,13 +210,17 @@ func CheckGoldenJSONResponse(t *testing.T, dir string, name string, dr *backend.
 	fpath := path.Join(dir, name+".jsonc")
 
 	expected, err := readGoldenJSONFile(fpath)
-	assert.NoError(t, err)
 	if err != nil {
-		return
+		if updateFile {
+			err = writeGoldenJSONFile(fpath, dr)
+			require.NoError(t, err)
+			return
+		}
+		require.NoError(t, err)
```

Comment:
Maybe `require.Fail` would make the most sense to use here IE

```go
require.Fail(t, "Error reading golden JSON file")
```

Reply:
added with [80ba379](https://github.com/grafana/grafana-plugin-sdk-go/pull/587/commits/80ba379be2a93b14f27f2864b15bd12f1593c722)